src/providers/multiwallet/SolanaConnector.ts

```diff
@@ -50,11 +50,17 @@ export class SolanaConnector
     this.connection = new Connection(this.clusterURL);
     this.providerURL = providerURL;
     this.emitter = new ConnectorEmitter<SolanaProvider, string>(debug);
-    this.wallet = new Wallet(this.providerURL, this.clusterURL);
 
-    // this.interval = setInterval(() => {
-    //   this.handleUpdate();
-    // }, 3000);
+    if (typeof this.providerURL === "string"){
+      this.wallet = new Wallet(this.providerURL, this.clusterURL);
+    } else if((this.providerURL as any).isBraveWallet) {
+      // faking postMessage method so that serum isInjectedProvider
+      // pass on Brave integrated solana wallet
+      this.providerURL.postMessage = () => {}
+      this.wallet = new Wallet(this.providerURL, this.clusterURL);
+    } else {
+      this.wallet = new Wallet(this.providerURL, this.clusterURL);
+    }
```

Comment:
The first and third cases are the same so is there any reason we couldn't do something like:

```suggestion
    if ((this.providerURL as any).isBraveWallet) {
      // faking postMessage method so that serum isInjectedProvider
      // pass on Brave integrated solana wallet
      this.providerURL.postMessage = () => {}
      this.wallet = new Wallet(this.providerURL, this.clusterURL);
    } else {
      this.wallet = new Wallet(this.providerURL, this.clusterURL);
    }
```